rpc/server.go

```diff
@@ -581,14 +599,18 @@ func (e *serverError) ErrorCode() string {
 	return codeNotImplemented
 }
 
-type nopObserverFactory struct{}
-
-func (nopObserverFactory) RPCObserver() Observer {
-	return nopObserver{}
+func ensureFactory(f RecorderFactory) RecorderFactory {
```

Comment:
There's a few places where we pass in nil for the factory.
We should always use ensureFactory?

Reply:
The methods that can be passed a nil recorder (Serve and ServeRoot) both call serve which calls ensureFactory before assigning it.